gluten-core/src/main/scala/io/glutenproject/execution/BasicPhysicalOperatorTransformer.scala

```diff
@@ -250,36 +229,20 @@ case class ProjectExecTransformer(projectList: Seq[NamedExpression],
 
   override def supportsColumnar: Boolean = GlutenConfig.getConf.enableColumnarIterator
 
-  override def doValidateInternal(): Boolean = {
+  override protected def doValidateInternal(): ValidationResult = {
     val substraitContext = new SubstraitContext
     // Firstly, need to check if the Substrait plan for this operator can be successfully generated.
     val operatorId = substraitContext.nextOperatorId(this.nodeName)
-    val relNode = try {
-      getRelNode(
-        substraitContext, projectList, child.output, operatorId, null, validation = true)
-    } catch {
-      case e: Throwable =>
-        this.appendValidateLog(s"Validation failed for ${this.getClass.toString}" +
-          s" due to: ${e.getMessage}")
-        return false
-    }
+    val relNode = getRelNode(
```

Comment:
I think this change is not correct. It will cause: 1) failure of translation of plan node is not caught. 2) should return directly as this step indicates the validation failed.

Reply:
> should return directly as this step indicates the validation failed.

If `getRelNode` throw exception, it would be caught by caller side (`doValidate`). Isn't it return directly for you ?

>  failure of translation of plan node is not caught

plan node is caught as we would inject unsupport tansformable hint to that plan, and before logging, it would add plan node name.